client/app/tasks/hooks/useTaskDetailsModal.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react'
+import { useRouter, useSearchParams } from 'next/navigation'
+import { TaskTypeWithUser } from '@/_types'
+import { CustomEventType } from '../types'
+
+export const useTaskDetailsModal = (tasks: TaskTypeWithUser[] | undefined) => {
+  const router = useRouter()
+  const searchParams = useSearchParams()
+  const taskId = searchParams.get('taskId')
+
+  const [selectedTask, setSelectedTask] = useState<TaskTypeWithUser | null>(
+    null,
+  )
+  const [isModalOpen, setIsModalOpen] = useState(false)
+
+  useEffect(() => {
+    if (taskId) {
+      const task = tasks?.find((task) => task.id === Number(taskId))
+      if (task) {
+        setSelectedTask(task)
+        setIsModalOpen(true)
+      }
+    }
+  }, [tasks, taskId])
+
+  const onModalClose = () => {
+    if (taskId) {
+      router.replace('/tasks', { scroll: false })
```

Comment:
Same. I wanted to delete the query if a user closes the modal because if you reload the page, you'll see the modal again. 
https://github.com/HAK2024/fairy-share/pull/114#discussion_r1559992137